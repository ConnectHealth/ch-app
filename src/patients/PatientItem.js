// @flow
import React from 'react';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import ListItem from 'material-ui/List/ListItem';
import IconButton from 'material-ui/IconButton';
import { colors } from 'material-ui/styles';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import type { Patient } from './store';

const PatientItem = ({ patient } : { patient: Patient}) => {
  const iconButtonElement = (
    <IconButton>
      <MoreVertIcon color={colors.grey400} />
    </IconButton>
  );
  const rightIconMenu = (
    <IconMenu iconButtonElement={iconButtonElement}>
      <MenuItem primaryText="Edit" />
      <MenuItem primaryText="Delete" />
    </IconMenu>
  );

  const text = `${patient.firstName} ${patient.lastName}`;

  return (
    <ListItem
      primaryText={text}
      rightIconButton={rightIconMenu}
    />
  );
};

export default PatientItem;
