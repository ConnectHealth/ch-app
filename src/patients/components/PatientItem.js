import React, { Component, PropTypes } from 'react';
import { ListItem, IconButton } from 'material-ui';
import { colors } from 'material-ui/styles';

import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

class PatientItem extends Component {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    const { patient, patientId } = this.props;

    // const deleteFn = () => deleteTodo(todoId);
    const nothing = () => {};
    const iconButtonElement = (
      <IconButton>
        <MoreVertIcon color={colors.grey400} />
      </IconButton>
    );
    const rightIconMenu = (
      <IconMenu iconButtonElement={iconButtonElement}>
        <MenuItem primaryText="Edit" onTouchTap={nothing()} />
        <MenuItem primaryText="Delete" onTouchTap={nothing()} />
      </IconMenu>
    );

      // const onTouchTap = () => complete(todoId);
    // TODO - this will link

    const text = `${patient.firstName} ${patient.lastName}`;

    return (
      <ListItem
        primaryText={text}
        rightIconButton={rightIconMenu}
      />
            // onTouchTap={onTouchTap}
            // onClick={onTouchTap}
      );
  }
}

PatientItem.propTypes = {
  patient: PropTypes.object.isRequired,
  patientId: PropTypes.number.isRequired,
};

export default PatientItem;
