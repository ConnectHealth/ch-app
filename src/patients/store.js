// @flow
import { action, observable, computed } from 'mobx';

/* import ApolloClient from 'apollo-client'; // default?*/

export class Patient {
  id = Math.floor(Math.random() * 100000).toString();
  @observable firstName = '';
  @observable lastName = '';
  constructor(firstName: string, lastName: string) {
    this.firstName = firstName;
    this.lastName = lastName;
  }
}

// FIXME - move
/* const allPatientsQuiery = gql`
 *
 * `;
 * */
export class PatientsStore {
  @observable searchText = '';
  @observable searching = false;
  @observable patients: Array<Patient> = // FIXME - use fetch
    [
      /* new Patient('John'),*/
      /* new Patient('John', 'Smith'),*/
      /* {
       *   id: 1,
       *   firstName: 'John',
       *   lastName: 'Smith',
       * },*/
      /* {
       *   id: 2,
       *   firstName: 'Sarah',
       *   lastName: 'Brown',
       * },*/
    ];

  @computed get searchResults(): Array<Patient> {
    return this.patients.filter(
      patient => patient.firstName.startsWith(this.searchText)
    );
  }

  @action search(text: string) {
    this.searchText = text;
  }

  @action create(firstName: string, lastName: string) {
    this.patients.push(new Patient(firstName, lastName));
  }

//  @action fetchPatients() {
//    this.searching = true;
//
//    Use superagent
//    this.patients =
//      [
//        new Patient('John', 'Smith'),
//        /* [
//         *   {
//         *     id: 1,
//         *     firstName: 'John',
//         *     lastName: 'Smith',
//         *   },
//         *   {
//         *     id: 2,
//         *     firstName: 'Sarah',
//         *     lastName: 'Brown',
//         *   },*/
//      ];
//    this.searching = false;
//  }
}

export default new PatientsStore();
