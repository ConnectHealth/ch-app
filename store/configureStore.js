import { createStore } from 'redux';
import rootReducer from '../layout/reducer';

export default function configureStore(initialState) {
  const store = createStore(rootReducer, initialState,
    window.devToolsExtension ? window.devToolsExtension() : undefined
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../layout/reducer', () => {
      const nextReducer = require('../layout/reducer');
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
