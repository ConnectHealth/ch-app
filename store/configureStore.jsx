import { createStore } from 'redux';
import rootReducer from '../layout/reducer';

export default function configureStore(initialState) {
  const store = createStore(rootReducer, initialState);

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../layout/reducer', () => {
      const nextReducer = require('../layout/reducer');
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
