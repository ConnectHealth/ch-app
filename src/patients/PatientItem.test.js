// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import PatientItem from './PatientItem';
import { Patient } from './store';

test('simple html', () => {
  const patient = new Patient(
    'first',
    'last',
  );

  const component = renderer.create(
    <MuiThemeProvider>
      <PatientItem patient={patient} />
    </MuiThemeProvider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('shallow', () => {
  const patient = new Patient(
    'first',
    'last',
  );

  const component = shallow(
    <PatientItem patient={patient} />,
  );

  expect(component.text()).toEqual('<ListItem />');
});

test('dom', () => {
  const patient = new Patient(
    'first',
    'last',
  );

  const component = mount(
    <MuiThemeProvider>
      <PatientItem patient={patient} />
    </MuiThemeProvider>,
  );

  expect(component.text()).toEqual('first last');
});

test('enzyme render', () => {
  const patient = new Patient(
    'first',
    'last',
  );

  const component = render(
    <MuiThemeProvider>
      <PatientItem patient={patient} />
    </MuiThemeProvider>,
  );

  expect(component.text()).toEqual('first last');
});
