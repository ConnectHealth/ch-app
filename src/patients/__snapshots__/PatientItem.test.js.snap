// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple html 1`] = `
<div>
  <span
    disabled={undefined}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    open={null}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "backgroundColor": null,
        "border": 10,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "fontWeight": "inherit",
        "lineHeight": "16px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "zIndex": 1,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "marginLeft": 0,
            "muiPrepared": true,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 56,
            "paddingTop": 16,
            "position": "relative",
          }
        }
      >
        <div
          className={undefined}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "display": "block",
              "muiPrepared": true,
              "position": "absolute",
              "right": 4,
              "top": 0,
            }
          }
        >
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "relative",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
                "zIndex": 1,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "#bdbdbd",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </div>
          </button>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "muiPrepared": true,
            }
          }
        >
          first last
        </div>
      </div>
    </div>
  </span>
</div>
`;
